### ISSUE TITLE
Diff parsing problem

### ISSUE THREAD

**Issue by bernhardreiter** [ISDuBA/ISDuBA#172]
@tschmidtb51 found a parsing or display problem for the side-by-side diff display:

![image](https://github.com/ISDuBA/ISDuBA/assets/8190008/47eb1659-8cfd-4f97-9c34-2ee809cf2a57)

The characters like `[-` and `-]` are probably not in the original files.

To compare, here is the integrated diff for the same section:
![Screenshot 2024-06-13 at 14-48-37 ICSA-23-108-02](https://github.com/ISDuBA/ISDuBA/assets/8190008/77308f83-ca13-4ea8-b469-d0ba8bbf1450)

**Comment by s-l-teichmann** [comment]
The problem may arise from the delimiters `{+`, `+}`,  `[-`  and `-]`.  I've took the idea of choose these symbols from the `--word-diff=plain` mode from Git.

Without further escaping this is not robust against containing delimiters inside a section.

@cintek 
I therefore propose to let a JSON representation do the segmentation part for us. Instead of a single string let us use a list of commands like `[{'m': 'i', 't': 'foo'}, {'m': 'o', 't': 'bar'}, {'m': 'd', 't': 'baz'}]` with `m` for **mode** and `t` for **text**, `i` for **insert**, `d` for **delete** and `o` for **original**.

This is be more verbose, but it should make the parsing easier/more robust and aligns well with the the `add`, `delete` and `insert` sections idea of the JSONPatch stuff.

**Comment by cintek** [comment]
@s-l-teichmann indeed, this looks more robust than the previous solution and is easy to implement in Svelte so I agree to do it this way.
